src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java

```diff
@@ -3,34 +3,58 @@
-
 	private final DriverManager driverManager;
-	private final static String figureJoe1 = "Figure Joe 1";
-	private final static String figureJoe2 = "Figure Joe 2";
-	private final static String figureJane1 = "Figures Jane";
 
 	public SelectTestFigureOptionListener(DriverManager driverManager) {
 		this.driverManager = driverManager;
 	}
 
 	@Override
 	public void actionPerformed(ActionEvent e) {
-		switch (e.getActionCommand()) {
-			case figureJoe1:
-				FiguresJoe.figureScript1(driverManager.getCurrentDriver());
-				break;
-			case figureJoe2:
-				FiguresJoe.figureScript2(driverManager.getCurrentDriver());
-				break;
-			case figureJane1:
-				FiguresJane.figureScript(new Job2dDriverAbstractDriverAdapter(driverManager.getCurrentDriver()));
-				break;
+		if (Figures.FIGURE_JOE_1.getFigureName().equals(e.getActionCommand())) {
+			FiguresJoe.figureScript1(driverManager.getCurrentDriver());
+		} else if (Figures.FIGURE_JOE_2.getFigureName().equals(e.getActionCommand())) {
+			FiguresJoe.figureScript2(driverManager.getCurrentDriver());
+		} else if (Figures.FIGURE_JANE.getFigureName().equals(e.getActionCommand())) {
+			FiguresJane.figureScript(new Job2dDriverAbstractDriverAdapter(driverManager.getCurrentDriver()));
+		} else if (Figures.FIGURE_CUS.getFigureName().equals(e.getActionCommand())) {
+			DriverCommand driverCommand = new SetPositionCommand(0, 0);
+			DriverCommand driverCommand1 = new OperateToCommand(0, 100);
+			DriverCommand driverCommand2 = new OperateToCommand(100, 100);
+			DriverCommand driverCommand3 = new OperateToCommand(100, 0);
+			DriverCommand driverCommand4 = new OperateToCommand(0, 0);
+
+			driverCommand.execute(driverManager.getCurrentDriver());
+			driverCommand1.execute(driverManager.getCurrentDriver());
+			driverCommand2.execute(driverManager.getCurrentDriver());
+			driverCommand3.execute(driverManager.getCurrentDriver());
+			driverCommand4.execute(driverManager.getCurrentDriver());
+		} else if (Figures.FIGURE_CUS2.getFigureName().equals(e.getActionCommand())) {
+			ComplexCommand complexCommand = new ComplexCommand();
+			complexCommand.addCommand(new SetPositionCommand(-100, 0));
+			complexCommand.addCommand(new OperateToCommand(100, 0));
+			complexCommand.addCommand(new OperateToCommand(100, 100));
+			complexCommand.addCommand(new OperateToCommand(-100, 100));
+			complexCommand.addCommand(new OperateToCommand(-100, 0));
+			complexCommand.execute(driverManager.getCurrentDriver());
+		} else if (Figures.FIGURE_CUS3.getFigureName().equals(e.getActionCommand())) {
+			ComplexCommand complexCommand1 = new ComplexCommand();
+			complexCommand1.addCommand(new SetPositionCommand(-100, 0));
+			complexCommand1.addCommand(new OperateToCommand(50, 100));
+			complexCommand1.addCommand(new OperateToCommand(100, 0));
+			complexCommand1.addCommand(new OperateToCommand(-100, 0));
+			complexCommand1.execute(driverManager.getCurrentDriver());
```

Comment:
This is not a factory. A factory should have methods returns a driver command (not execute it). Missing point 3.1.5.